src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py

```diff
@@ -1739,6 +1739,7 @@ def create_vmss(vmss_name, resource_group_name, image,
     # determine final defaults and calculated values
     tags = tags or {}
     os_disk_name = os_disk_name or 'osdisk_{}'.format(hash_string(vmss_id, length=10))
+    backend_pool_name = backend_pool_name or '{}BEPool'.format(load_balancer or application_gateway)
```

Comment:
You know, this extraction will have a side effect, that the default `backend_pool_name` will be changed, say from 'foo1`LBBEPool`' to 'foo1`BEPool`'. When the balancer type is `new`, `backend_pool` will always be created, this change is fine, otherwise we need to reevaluate